include/verilated_vcd_c.h

```diff
@@ -48,48 +47,26 @@ class VerilatedVcdFile {
     virtual ssize_t write(const char* bufp, ssize_t len) VL_MT_UNSAFE;
 };
 
-//=============================================================================
-// VerilatedVcdSig
-/// Internal data on one signal being traced.
-
-class VerilatedVcdSig {
-protected:
-    friend class VerilatedVcd;
-    vluint32_t m_code;  ///< VCD file code number
-    int m_bits;  ///< Size of value in bits
-    VerilatedVcdSig(vluint32_t code, int bits)
-        : m_code(code)
-        , m_bits(bits) {}
-
-public:
-    ~VerilatedVcdSig() {}
-};
-
-//=============================================================================
-
-typedef void (*VerilatedVcdCallback_t)(VerilatedVcd* vcdp, void* userthis, vluint32_t code);
-
 //=============================================================================
 // VerilatedVcd
 /// Base class to create a Verilator VCD dump
 /// This is an internally used class - see VerilatedVcdC for what to call from applications
 
-class VerilatedVcd {
+class VerilatedVcd : public VerilatedTrace<VerilatedVcd> {
 private:
+    // Give the superclass access to private bits (to avoid virtual functions)
+    friend VerilatedTrace<VerilatedVcd>;
```

Comment:
Pre C11:

include/verilated_vcd_c.h:58: error: a class-key must be used when declaring a friend
include/verilated_vcd_c.h:58: error: friend declaration does not name a class or function

Reply:
Hopefully what I just pushed should fix that